libxrdp/xrdp_caps.c

```diff
@@ -612,17 +653,23 @@ xrdp_caps_process_confirm_active(struct xrdp_rdp *self, struct stream *s)
     int len;
     char *p;
 
-    LOG_DEVEL(LOG_LEVEL_TRACE, "in xrdp_caps_process_confirm_active");
     in_uint8s(s, 4); /* rdp_shareid */
     in_uint8s(s, 2); /* userid */
     in_uint16_le(s, source_len); /* sizeof RDP_SOURCE */
     in_uint16_le(s, cap_len);
     in_uint8s(s, source_len);
     in_uint16_le(s, num_caps);
     in_uint8s(s, 2); /* pad */
+    LOG_DEVEL(LOG_LEVEL_TRACE, "Received [MS-RDPBCGR] TS_CONFIRM_ACTIVE_PDU "
+              "shareID (ignored), originatorID (ignored), lengthSourceDescriptor %d, "
+              "lengthCombinedCapabilities  %d, sourceDescriptor (ignored), "
+              "numberCapabilities %d", source_len, cap_len, num_caps);
 
     if ((cap_len < 0) || (cap_len > 1024 * 1024))
     {
+        LOG_DEVEL(LOG_LEVEL_ERROR, "Received [MS-RDPBCGR] TS_CONFIRM_ACTIVE_PDU "
```

Comment:
I'm happy to make all of the error messages `LOG()` instead of `LOG_DEVEL()`.
I have noticed that I when I enable all of the tracing for this library with `LOG_DEVEL()` enabled, there is a significant performance penalty. The performance penalty is likely from all the writing to the file and console, but I haven't checked that yet. I've been err-ing on the side of using `LOG_DEVEL()` unless it's obvious that the log message is for a sysadmin. I'm happy to use `LOG()` for any log message that is suggested.

Reply:
I found one possible reason for the performance penalty when I was working on the redirection stuff a while ago - the file based logging uses locking between messages because the logs are shared between processes and their children. This seems to be expensive. If you disable the file logging you might find it runs a bit better. If that's the case we can discuss solutions.

As regards the LOG()/LOG_DEVEL() thing, all I'm being guided by is that an error which will result in session disconnection should be logged by LOG(ERROR,), so that we've got something to go on in fault reports. It's a simple rule we could potentially add to the logging guidelines.

Arguments against this would be:-
- It's entirely possible for some of the suggestions I've made that a perfectly good message has been generated at a lower level in the stack.
- What you've added here is so much better than what we've got that maybe I'm being too picky

There are probably others. From what you've added in some of your messages it looks like you've got a much better idea than I have about exactly what's going on in much of this code.

Reply:
I agree that error logging should all use `LOG`. I've changed all the error logging statements in the latest update.

I've updated the log levels section of the [Logging wiki](https://github.com/neutrinolabs/xrdp/wiki/Logginghttps://github.com/neutrinolabs/xrdp/wiki/Logging) page to mention that all error logs should be logged using `LOG(ERROR)`.

Reply:
Excellent - thanks.